src/Features/CSharp/Portable/Snippets/CSharpForLoopSnippetProvider.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Snippets;
+using Microsoft.CodeAnalysis.Snippets.SnippetProviders;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Snippets
+{
+    [ExportSnippetProvider(nameof(ISnippetProvider), LanguageNames.CSharp), Shared]
+    internal sealed class CSharpForLoopSnippetProvider : AbstractForLoopSnippetProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpForLoopSnippetProvider()
+        {
+        }
+
+        /// <summary>
+        /// Creates the for loop statement syntax.
+        /// Must be done in language specific file since there is no generic way to generate the syntax.
+        /// </summary>
+        protected override async Task<SyntaxNode> CreateForLoopStatementSyntaxAsync(Document document, int position, CancellationToken cancellationToken)
+        {
+            var compilation = await document.Project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var iteratorName = NameGenerator.GenerateUniqueName(
+                new List<string> { "i", "j", "k", "a", "b", "c" },
+                n => semanticModel.LookupSymbols(position, name: n).IsEmpty);
+            var indexVariable = generator.Identifier(iteratorName);
+
+            // Creating the variable declaration based on if the user has
+            // 'var for built in types' set in their editorconfig.
+            var variableDeclarationSyntax =
+                 SyntaxFactory.VariableDeclaration(compilation.GetSpecialType(SpecialType.System_Int32).GenerateTypeSyntax(allowVar: false),
```

Comment:
Why `allowVar` is false here? Is it possible to to link this to the editorconfig value here
https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.CodeStyle/CSharpCodeStyleOptions.cs,cb8ef73fcae9e59f,references

Reply:
> Why allowVar is false here? 

Yeah it should be true.

---

> Is it possible to to link this to the editorconfig value here

`WithAdditionalAnnotations(Simplifier.Annotation)` is needed. That way the code cleanup will be responsible of respecting the user preference. This should happen here:

https://github.com/dotnet/roslyn/blob/35e54c1b711eb741eb4d4cc1b133e09d79e17e7d/src/Features/Core/Portable/Snippets/SnippetProviders/AbstractSnippetProvider.cs#L108